app/controllers/live_controller.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class LiveController < ApplicationController
+  def admin
+    @competition_id = params[:competition_id]
+    @competition = Competition.find(@competition_id)
+    @round = Round.find(params[:round_id])
+    @event_id = @round.event.id
+    @competitors = @round.registrations_with_wcif_id
+  end
+
+  def add_result
+    results = params.require(:attempts)
+    round_id = params.require(:round_id)
+    registration_id = params.require(:registration_id)
+
+    if LiveResult.exists?(round_id: round_id, registration_id: registration_id)
+      return render json: { status: "result already exist" }, status: :unprocessable_entity
+    end
+
+    AddLiveResultJob.perform_now({ results: results,
+                                   round_id: round_id,
+                                   registration_id: registration_id,
+                                   entered_by: current_user })
+
+    render json: { status: "ok" }
+  end
+
+  def update_result
+    results = params.require(:attempts)
+    round = Round.find(params.require(:round_id))
+    registration_id = params.require(:registration_id)
+
+    result = LiveResult.includes(:live_attempts).find_by(round: round, registration_id: registration_id)
+
+    unless result.present?
+      return render json: { status: "result does not exist" }, status: :unprocessable_entity
+    end
+
+    previous_attempts = result.live_attempts
+
+    new_attempts = results.map.with_index(1) do |r, i|
+      same_result = previous_attempts.find_by(result: r, attempt_number: i)
+      if same_result.present?
+        same_result
+      else
+        different_result = previous_attempts.find_by(attempt_number: i)
+        new_result = LiveAttempt.build(result: r, attempt_number: i)
+        different_result&.update(replaced_by_id: new_result.id)
+        new_result
+      end
+    end
```

Comment:
Note to self: I have a hunch there might be an even more efficient way to handle this.

Reply:
Will play around later when I have more headspace.